### PR TITLE
Added Angular guide

### PR DESCRIPTION
**Pull request by engineering-this**
Added angular guide, which partially solves https://github.com/ckeditor/ckeditor-dev/issues/2920.

### CODE REVIEW

File: docs/guide/dev/integration/angular/README.md [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+Copyright (c) 2003-2019, CKSource - Frederico Knabben. All rights reserved.
+For licensing, see LICENSE.md.
+-->
+
+# Angular Integration
+
+<info-box info=""> This feature is provided through the <a href="https://www.npmjs.com/package/ckeditor4-angular"><code>ckeditor4-angular</code> npm package</a>.
+</info-box>
+
+CKEditor offers native Angular integration through `CKEditor` Angular component. It provides deep integration of CKEditor and Angular that lets you use the native features of CKEditor inside Angular component. The `CKEditor` Angular component is compatible with Angular versions 2.0+.
```
Comment (mlewand) [inline docs/guide/dev/integration/angular/README.md]:
```suggestion
CKEditor offers native Angular integration through `CKEditor` Angular component. It provides deep integration of CKEditor and Angular that lets you use the native features of CKEditor inside Angular component. The `CKEditor` Angular component is compatible with Angular versions 2.0 and later.
```

File: docs/guide/dev/integration/angular/README.md [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+
+## Customizing CKEditor Preset or Version
+
+By default `CKEditor` Angular component loads standard preset of the latest CKEditor release from <a href="https://cdn.ckeditor.com/">CDN</a> while creating the first editor. This behavior can be altered by setting the value of `editorUrl` attribute in template to point to desired CKEditor location:
+
+```html
+<ckeditor [editorUrl]="https://your-website.example/ckeditor/ckeditor.js"></ckeditor>
+```
+
+Note that attribute has to be assigned **before first component is initialized**.
```
Comment (mlewand) [inline docs/guide/dev/integration/angular/README.md]:
```suggestion
Note that attribute has to be assigned **before the first component is initialized**.
```

File: docs/guide/dev/integration/angular/README.md [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+
+``` typescript
+
+import { CKEditor4 } from "@ckeditor/ckeditor4-angular/ckeditor";
+
+@Component( {
+    ...
+} )
+export class MyComponent {
+    public Editor = ClassicEditor;
```
Comment (mlewand) [inline docs/guide/dev/integration/angular/README.md]: I don't understand this code, ClassicEditor is not defined.

File: docs/guide/dev/integration/angular/README.md [hunk 4]
```diff
@@ -0,0 +1,241 @@
...
+Alternatively you can load CKEditor before loading `CKEditor` Angular component. In that case the component will use the already loaded CKEditor:
+
+```html
+<script src="custom-ckeditor.js"></script>
+<script src="main.js"></script>
+```
+
+## Choosing Editor Type
+
+By default `CKEditor` Angular component creates {@link guide/dev/framed/README classic editor}. To create {@link guide/dev/inline/README inline editor} `type` property with value of `inline` must be added:
```
Comment (mlewand) [inline docs/guide/dev/integration/angular/README.md]: From what I know, currently there's no support for classic editor in our angular integration. Is this section relevant to this integration?

**Comment by engineering-this** [comment]
Pushing changes:
- Rephrased _editor types_ section, to reflect actual supported editor types.
- Corrected few code samples, removed redundant declarations/attributes.
- Corrected other requested issues, grammar, etc.

File: docs/guide/dev/integration/angular/README.md [hunk 5]
```diff
@@ -0,0 +1,245 @@
...
+
+The following `@Input` properties are supported by CKEditor 4 Angular component:
+
+### `ready`
+
+Fires when the editor is ready. It corresponds with the `editor#instanceReady` https://ckeditor.com/docs/ckeditor4/latest/api/CKEDITOR_editor.html#event-instanceReady event.
+
+### `change`
+
+Fires when the content of the editor has changed. In contrast to `change` - only emits when data really changed thus can be successfully used with `[data]` and two way `[(data)]` binding.
```
Comment (mlewand) [inline docs/guide/dev/integration/angular/README.md]: "Fires when the content of the editor has changed." is duplicated **three** times, this makes it extremely redundant.

File: docs/guide/dev/integration/angular/README.md [hunk 6]
```diff
@@ -0,0 +1,245 @@
...
+    public onChange( event: CKEditor4.EventInfo ) {
+        console.log( event.editor.getData() );
+    }
+    ...
+}
+```
+
+### `dataChange`
+
+Fires when the content of the editor has changed. In contrast to `change` - only emits when data really changed thus can be successfully used with `[data]` and two way `[(data)]` binding.
```
Comment (mlewand) [inline docs/guide/dev/integration/angular/README.md]: This is an exact copy of "Fires when the content of the editor has changed. In contrast to change - only emits when data really changed thus can be successfully used with [data] and two way [(data)] binding." from a previous section.

File: docs/guide/dev/integration/angular/README.md (hunk 5 above)
Comment (mlewand) [reply docs/guide/dev/integration/angular/README.md]: How `change` event can work "In contrast to `change` "? `change` is the event that is described here.

File: docs/guide/dev/integration/angular/README.md [hunk 7]
```diff
@@ -0,0 +1,245 @@
...
+
+### `focus`
+
+Fires when the editing view of the editor is focused. It corresponds with the `editor#focus` https://ckeditor.com/docs/ckeditor4/latest/api/CKEDITOR_editor.html#event-focus event.
```
Comment (mlewand) [inline docs/guide/dev/integration/angular/README.md]: Most common name for this is editable, let's stick to it.

```suggestion
Fires when the editor's editable is focused. It corresponds with the `editor#focus` https://ckeditor.com/docs/ckeditor4/latest/api/CKEDITOR_editor.html#event-focus event.
```

Same change for the blur event.

Comment (mlewand) [reply docs/guide/dev/integration/angular/README.md]: It should be a link, instead there's a event name followed by a full URL. Same thing for blur.

File: docs/guide/dev/integration/angular/README.md [hunk 8]
```diff
@@ -0,0 +1,245 @@
...
+
+```typescript
+component.instance.getData();
+```
+
+Please note that this property is initialised asynchronously, when component is ready.
+
+## Types
+
+CKEditor types can be installed from [@types/ckeditor](https://www.npmjs.com/package/@types/ckeditor). Note, they aren't developed by the CKEditor Team, so we can't guarantee that are up to date.
```
Comment (mlewand) [inline docs/guide/dev/integration/angular/README.md]:
```suggestion
CKEditor types can be installed from [@types/ckeditor](https://www.npmjs.com/package/@types/ckeditor). Note, they aren't maintained by the CKEditor Team, so we can't guarantee that are up to date.
```

File: docs/guide/dev/integration/angular/README.md [hunk 9]
```diff
@@ -0,0 +1,245 @@
...
+<ckeditor [editorUrl]="https://your-website.example/ckeditor/ckeditor.js"></ckeditor>
+```
+
+Note that attribute has to be assigned **before the first component is initialized**.
+
+Alternatively you can load CKEditor before loading `CKEditor` Angular component. In that case the component will use the already loaded CKEditor:
+
+```html
+<script src="custom-ckeditor.js"></script>
+<script src="main.js"></script>
```
[the remark this answers is not in the document]
Comment (engineering-this) [reply docs/guide/dev/integration/angular/README.md]: I mean bundled Angular package, as it has to be loaded after the `ckeditor.js`. What do you think, about adding a comment:

```suggestion
<script src="main.js"></script><!–– Your bundled Angular package ––>
```
or
```suggestion
<script src="your-app-bundle.js"></script>
```
or just drop this line?

[the remark this answers is not in the document]
Comment (mlewand) [reply docs/guide/dev/integration/angular/README.md]: Both propositions look good. I think the second is cleaner 👌

File: docs/guide/dev/integration/angular/README.md (hunk 6 above)
Comment (engineering-this) [reply docs/guide/dev/integration/angular/README.md]: I left this section as it correct and updated `change`.

**Comment by engineering-this** [comment]
Additionally I've altered some links, to use `{@linkapi}` or `{@link}`.

File: docs/guide/dev/integration/angular/README.md [hunk 10]
```diff
@@ -0,0 +1,235 @@
...
+```
+
+The `data` attribute used in the above example is responsible for setting the editor's data.
+
+## Customizing CKEditor Preset or Version
+
+By default `CKEditor` Angular component loads [Standard Preset](https://ckeditor.com/docs/ckeditor4/latest/examples/standardpreset.html) of the latest CKEditor release from <a href="https://cdn.ckeditor.com/">CDN</a> while creating the first editor. This behavior can be altered by setting the value of `editorUrl` attribute in template to point to desired CKEditor location:
+
+```html
+<ckeditor [editorUrl]="https://your-website.example/ckeditor/ckeditor.js"></ckeditor>
```
Comment (mlewand) [inline docs/guide/dev/integration/angular/README.md]: Setting `editorUrl` attribute like that will throw exception:

```
compiler.js:15859 Uncaught Error: Quotes are not supported for evaluation!
        Statement: //cdn.ckeditor.com/4.5.5/standard-all/ckeditor.js located at ng:///AppModule/AppComponent.html@3:11
```

Comment (mlewand) [reply docs/guide/dev/integration/angular/README.md]: Should be without brackets.

File: docs/guide/dev/integration/angular/README.md [hunk 11]
```diff
@@ -0,0 +1,235 @@
...
+URL address to `ckeditor.js`, for more detailed information see [Customising CKEditor Preset or Version](#customizing-ckeditor-preset-or-version).
+
+### `config`
+
+Custom configuration can be passed to the editor via `config` attribute passed to `<ckeditor>` in component template. The following example shows {@link guide/dev/features/toolbar/README how to change the contents of the toolbar}:
+
+```html
+<ckeditor
+	data="<p>Editor' content</p>"
+	[config]="{ { toolbar: [ [ 'Bold' ] ] } }"
```
Comment (mlewand) [inline docs/guide/dev/integration/angular/README.md]: Value of `[config]` throws parsing error to the console.

File: docs/guide/dev/integration/angular/README.md [hunk 12]
```diff
@@ -0,0 +1,235 @@
...
+
+The following `@Output` properties are supported by CKEditor 4 Angular component:
+
+### `ready`
+
+Fires when the editor is ready. It corresponds with the {@linkapi CKEDITOR.editor#instanceReady `editor#instanceReady`} event.
+
+### `change`
+
+Fires when the content of the editor has changed. It corresponds with the {@linkapi CKEDITOR.editor#change `editor#change`} event. For performance reasons this event may be called even when data didn't really changed.
```
[the remark this answers is not in the document]
Comment (engineering-this) [reply docs/guide/dev/integration/angular/README.md]: Components `change` fully matches https://ckeditor.com/docs/ckeditor4/latest/api/CKEDITOR_editor.html#event-change the rationale comes from our docs.

**Comment by engineering-this** [comment]
- Corrected code samples, they are all working correctly now.
- Fixed broken link.
- Removed ellipsis symbol from html samples, they didn't look good there, and `<ckeditor></ckeditor> work without any extra attributes anyway.

File: docs/guide/dev/integration/angular/README.md [hunk 13]
```diff
@@ -0,0 +1,234 @@
...
+
+```html
+<ckeditor (change)="onChange($event)"></ckeditor>
+```
+
+``` typescript
+
+import { CKEditor4 } from "@ckeditor/ckeditor4-angular/ckeditor";
```
Comment (mlewand) [inline docs/guide/dev/integration/angular/README.md]: We use single quotes for js strings.

File: docs/guide/dev/integration/angular/README.md [hunk 14]
```diff
@@ -0,0 +1,234 @@
...
+
+Fires when the editor's editable is focused. It corresponds with the {@linkapi CKEDITOR.editor#focus `editor#focus`} event.
+
+### `blur`
+
+Fires when the editing view of the editor is blurred. It corresponds with the {@linkapi CKEDITOR.editor.blur `editor#blur`} event.
+
+### Note
+
+Callbacks bound to each of events, are called with {@linkapi CKEDITOR.eventInfo `eventInfo`}. For importing and usage of `EventInfo` interface see [**`change`** example](#change).
```
Comment (mlewand) [inline docs/guide/dev/integration/angular/README.md]: Not each event, as [`dataChange` is fired with a different signature](https://github.com/cksource/ckeditor4-angular/blob/3bfa435/src/ckeditor/ckeditor.component.ts#L293) it gets a plain string.

Comment (mlewand) [reply docs/guide/dev/integration/angular/README.md]: I think it should be changed in the upstream, that is `ckeditor4-angular` integration.
